bitcoin/src/address/mod.rs

```diff
@@ -472,37 +497,83 @@ impl Address {
         network: Network,
     ) -> Address {
         let (output_key, _parity) = internal_key.tap_tweak(secp, merkle_root);
-        let prog = WitnessProgram::new(WitnessVersion::V1, output_key.to_inner().serialize())
-            .expect("taproot output key has len 32 <= 40");
-        let payload = Payload::WitnessProgram(prog);
-        Address(AddressInner { network, payload }, PhantomData)
+        let ser = output_key.to_inner().serialize();
+        let program = WitnessProgram::new(WitnessVersion::V1, ser)
+            .expect("serialized taproot key is a valid program");
+        Address::segwit(program, network)
     }
 
     /// Creates a pay to taproot address from a pre-tweaked output key.
     ///
     /// This method is not recommended for use, [`Address::p2tr()`] should be used where possible.
     pub fn p2tr_tweaked(output_key: TweakedPublicKey, network: Network) -> Address {
-        let prog = WitnessProgram::new(WitnessVersion::V1, output_key.to_inner().serialize())
-            .expect("taproot output key has len 32 <= 40");
-        let payload = Payload::WitnessProgram(prog);
-        Address(AddressInner { network, payload }, PhantomData)
+        let ser = output_key.to_inner().serialize();
+        let program = WitnessProgram::new(WitnessVersion::V1, ser)
+            .expect("serialized taproot output key has len 32 <= 40");
+        Address::segwit(program, network)
     }
 
     /// Creates an address from an arbitrary witness program.
     ///
     /// This only exists to support future witness versions. If you are doing normal mainnet things
     /// then you likely do not need this constructor.
     pub fn from_witness_program(program: WitnessProgram, network: Network) -> Address {
-        let inner = AddressInner { payload: Payload::WitnessProgram(program), network };
+        let hrp = KnownHrp::from_network(network);
+        let inner = AddressInner::Segwit { program, hrp };
         Address(inner, PhantomData)
     }
 
+    fn segwit(program: WitnessProgram, network: Network) -> Address {
+        let hrp = KnownHrp::from_network(network);
+        let inner = AddressInner::Segwit { program, hrp };
+        Self(inner, PhantomData)
+    }
```

Comment:
I can't see how this is different from `from_witness_program`.

Reply:
lol, and that function is directly above it in the code, OMG, that is some wild tunnel vision - I have been up since 2am coding :)

Will remove the `segwit` function.

Reply:
Hehe, similar things happen to me occasionally. And the signature/code match exactly. So I made it a habit to always check the doc before publishing issues/PRs.

Reply:
> So I made it a habit to always check the doc before publishing issues/PRs.

Oh yeah? Do you include private functions in the docs build and then just quickly scan the built docs of the modules you changed to see everything looks sane?

Reply:
I don't do private though.